src/commands/src/helpers/dependency-helpers.ts

```diff
@@ -33,12 +34,42 @@ async function createEnvProdFile(repoDir: string, projectName: string, baseUrl:
   }
 }
 
-export async function prepareBuildAndGetDependencies(repoName: string, projectName: string, baseUrl: string) {
+export async function downloadBuildDependencies(buildDependencies: BuildDependencies | null, repoName: string) {
```

Comment:
NIT: It might be simpler to read this if we invert the if statement conditional and return early:
```ts

if (!buildDependencies) return;

const commands: string[] = [];
buildDependencies.map((dependencyInfo) => {
    const repoDir = getRepoDir(repoName);
    const buildDir = dependencyInfo.buildDir ?? repoDir;
    executeCliCommand({
    command: 'mkdir',
    args: ['-p', buildDir],
    });
    commands.push(`mkdir -p ${buildDir}`);
    dependencyInfo.dependencies.map((dep) => {
    executeCliCommand({
        command: 'curl',
        args: ['-SfL', dep.url, '-o', `${buildDir}/${dep.filename}`],
    });
    commands.push(`curl -SfL ${dep.url} -o ${buildDir}/${dep.filename}`);
    });
});
return commands;
```

Or, it might be nice to make the `buildDependencies` object required, and check when we call the `downloadBuildDependencies` function that it is defined.